### ISSUE TITLE
Any plans to update with Instance normalization?

### ISSUE THREAD

**Issue by igorbarinov** [yusuketomoto/chainer-fast-neuralstyle#19]
Hi, do you have any plans to update your solution with Instance normalization as described here "Instance Normalization: The Missing Ingredient for Fast Stylization" https://arxiv.org/abs/1607.08022

**Comment by austingg** [comment]
@igorbarinov  what's the difference between instance nomalization and batchnorm with batchsize = 1 ?

**Comment by austingg** [comment]
@igorbarinov  got it. instance normalization only normalized in spatial not cross feature map and batch

**Comment by bucktoothsir** [comment]
@austingg Have you tried it? I think it does the same thing with batch normalization when you set batchsize = 1

**Comment by austingg** [comment]
@bucktoothsir  I haven't tried.  but batchsize = 1 still normalize cross the feature maps. instance normalization only do normalization within the feature map plane. As @yusuketomoto  say, batchsize >=2 have some problem, i do all my experiments with batchsize = 1

**Comment by austingg** [comment]
@bucktoothsir  sorry .  After i checked the paper, i found batchsize = 1 is instance normalization.

**Comment by ttoinou** [comment]
Are you sure ? According to the paper that would mean that using batchsize=1 yields better results than batchsize >= 2

**Comment by austingg** [comment]
@ttoinou  batch normalization is across instances, however, when batch size = 1, there is only one instance. You can also implement multi batch training with normaliztion within instance.

**Comment by logic1988** [comment]
From the paper we can see the effect is greatly improved. Anybody have the implement in python, I can test it.

**Comment by ttoinou** [comment]
@logic1988 : Yes they updated their project yesterday with instance normalization with torch ! https://github.com/DmitryUlyanov/texture_nets 
@austingg  : Great so there's still some hope to get the training faster by fixing the batchsize issue (I guess that loading images, resizing them and giving them to the GPU would be faster in batch, no ? there would be less CPU <-> GPU communication)

**Comment by logic1988** [comment]
There are a great Torch Implementation released by jcjohnson a few days ago.
https://github.com/jcjohnson/fast-neural-style

In addition to the original method, the new method inspired by the the paper [Instance Normalization: The Missing Ingredient for Fast Stylization] has been implemented.

The latter seems to have a better result and faster speed than the former.

It said that:

> These models use the same architecture as those used in our paper, except with half the number of filters per layer and with instance normalization instead of batch normalization. Using narrower layers makes the models smaller and faster without sacrificing model quality.

How to modify the corresponding code in this project? @yusuketomoto

**Comment by artnose** [comment]
On the same system, is the Torch implementation or this implementation faster?  better?  Anyone have comparisons?
